OPERATOR.rst

```diff
@@ -156,28 +156,50 @@ label from blocking PR. Last, remove the blocking relationship.
 Upgrading GitLab
 ^^^^^^^^^^^^^^^^
 
-Occasionally it falls on the operator to upgrade the Azul GitLab instance. If
-the current major version is ``n`` and the latest available major version is
-greater than ``n+1`` (i.e. upgrading directly to the latest version would skip
-one or more major versions) then multiple successive upgrades must be made, such
-that no upgrade skips a major version. For example, if the current version is
-13.x.y and the latest available version is 15.x.y, then one would first upgrade
-to 14.x.y and then repeat the process to upgrade to 15.x.y.
-
-Before any changes are applied, stop the instance (do not terminate) and create
-a snapshot of its EBS volume. Edit ``terraform/gitlab/gitlab.tf.json.template.py``,
-updating the versions of the docker images for ``gitlab-ce`` and
-``gitlab-runner``. Then run::
+Operators must check for updates to GitLab on a daily basis. Updates to Gitlab
+can be quickly checked at the ``/help`` endpoint for `GitLab dev`_ and
+`GitLab prod`_. If there are updates available, find the new release for the
+`GitLab Docker image`_ and update the `Gitlab Terraform`_ file. When updating
+the GitLab instance, check if there are applicable updates to the
+`GitLab runner image`_. Before starting the update process check the
+`GitLab release notes`_ for upgrading instructions. When upgrading across
+multiple GitLab versions, follow the prescribed GitLab `upgrade path`_.
+
+Before any changes are applied, checkout a new branch with `github/develop` as
+its start-point, run::
+
+	git fetch github
+	git checkout -b gitlab/<yyy-mm-dd> github/develop
+
+Stop the GitLab EC2 instance (do not terminate), and create a snapshot of its
+EBS volume.
```

Comment:
This, on the other hand, is not detailed enough. The snapshot should be tagged with an `owner` and have a specific name. Document the steps needed to do this.

Reply:
I added a script to help do this, creating multiple snapshots with tags and descriptions opens up room for operator errors. As the script is now it should add the tags, track the operator who created the snapshots, date-stamp the description to track the dates.

Currently it looks like operators are not manually adding tags (understandable as it's not documented, and easy to overlook) and the deletion daemon is deleting them.

Reply:
https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#SnapshotDetails:snapshotId=snap-06f05105a554d3557
for what the snapshot looks like.

FYI it looks like the `deployment_stage` is currently used when naming the gitlab instance, rather than `gitlab-azul-dev`, it's just called `gitlab-azul` not sure if this is intentional.

Reply:
https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#SnapshotDetails:snapshotId=snap-06f05105a554d3557 

This is what the script output looks like, is the deployment stage name intentionally removed from `azul-gitlab`? I would expect it to follow other infra naming conventions that we have in place where the stage is also visible e.g. `azul-gitlab-dev`

Reply:
No. The name should include the stage name.

Reply:
Added fixme for this https://github.com/DataBiosphere/azul/issues/3820

Reply:
Removed fixme for #3820, as the issue is closed.